client/state/application/reducer.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { CONNECTION_LOST, CONNECTION_RESTORED } from 'state/action-types';
+
+export function isOnline( state = 'CHECKING', action ) {
```

Comment:
I commented directly on the commit, but I think that can be easy to miss, so I'll mention it here. I think it would be best to rename this function or to change it to return a boolean. It seems inconsistent as it is.